File: core/identity-service/src/main/java/com/gov/crypto/controller/AdminController.java
package com.gov.crypto.controller;

import org.springframework.http.ResponseEntity;
import org.springframework.security.access.prepost.PreAuthorize;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.Map;

@RestController
@RequestMapping("/api/v1/admin")
@PreAuthorize("hasRole('ADMIN')")
// Implementation of Admin Stats for Dashboard
public class AdminController {

    @GetMapping("/stats")
    public ResponseEntity<Map<String, Object>> getAdminStats() {
        // Mock stats for Admin Dashboard
        return ResponseEntity.ok(Map.of(
                "totalUsers", 150,
                "activeCertificates", 45,
                "pendingRequests", 12,
                "signedToday", 28));
    }
}

File: backend/identity-service/src/main/kotlin/com/gov/crypto/identityservice/controller/AdminController.kt
package com.gov.crypto.identityservice.controller

import com.gov.crypto.identityservice.entity.KycStatus
import com.gov.crypto.identityservice.entity.Roles
import com.gov.crypto.identityservice.repository.UserRepository
import org.springframework.http.ResponseEntity
import org.springframework.security.access.prepost.PreAuthorize
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/admin")
class AdminController(private val userRepository: UserRepository) {

    data class KycApprovalRequest(val username: String, val action: String) // action: APPROVE or REJECT

    @PostMapping("/approve-kyc")
    @PreAuthorize("hasAnyRole('ADMIN', 'OFFICER')")
    fun approveKyc(@RequestBody request: KycApprovalRequest): ResponseEntity<String> {
        // Role check enforced by @PreAuthorize annotation
        
        val user = userRepository.findByUsername(request.username)
            .orElseThrow { RuntimeException("User not found") }

        if (request.action.uppercase() == "APPROVE") {
            user.kycStatus = KycStatus.APPROVED
        } else if (request.action.uppercase() == "REJECT") {
            user.kycStatus = KycStatus.REJECTED
        } else {
            return ResponseEntity.badRequest().body("Invalid action. Use APPROVE or REJECT")
        }
        
        userRepository.save(user)
        return ResponseEntity.ok("KYC status updated to ${user.kycStatus} for user ${request.username}")
    }
}
